### PR TITLE
Fixing the "This competition's results are visible but not posted"

### PR DESCRIPTION
**Pull request by lnzainn**
This pr should solve #9229 , hopefully.

@danieljames-dj @dunkOnIT this might have errors so please review it :)

### CODE REVIEW

**Review by danieljames-dj — COMMENTED** [review]
Thanks for the contribution Zain, left couple of comments. Regarding the test failure, there is a unit-test failure.

File: app/views/users/edit.html.erb [hunk 1]
```diff
@@ -109,7 +109,7 @@
             </div>
             <div class="col-xs-6">
               <%= f.input :wca_id, as: :wca_id, disabled: !editable_fields.include?(:wca_id)%>
-              <% if current_user.any_kind_of_delegate? && @user.is_special_account? %>
+              <% if (current_user.any_kind_of_delegate? || current_user.can_edit_user?(@user)) && @user.is_special_account? %>
```
[the remark this answers is not in the document]
Comment (lnzainn) [reply app/views/users/edit.html.erb]: I was creating a different PR and it accidentally got mixed with this

File: config/locales/en.yml [hunk 2]
```diff
@@ -1339,6 +1339,7 @@ en:
       uncancel_success: "Successfully uncancelled competition!"
       uncancel_failure: "Cannot uncancel the competition."
       results_not_posted: "This competition's results are visible but haven't been posted yet."
+      results_not_posted_non_admins: "We are busy processing this competition's results - they should be available shortly."
```
[the remark this answers is not in the document]
Comment (lnzainn) [reply config/locales/en.yml:1342]: @gregorbg do you have any suggestions for the key name?

[the remark this answers is not in the document]
Comment (dunkOnIT) [reply config/locales/en.yml:1342]: I suggest `results_still_processing`

[the remark this answers is not in the document]
Comment (dunkOnIT) [reply config/locales/en.yml:1342]: Gregor's plate is pretty full, hopefully he's fine for Daniel and I to resolve this :D

File: app/models/competition.rb [hunk 3]
```diff
@@ -469,7 +469,11 @@ def warnings_for(user)
       end
 
       if self.results.any? && !self.results_posted?
-        warnings[:results] = I18n.t('competitions.messages.results_not_posted')
+        if user.can_admin_results?
```
Comment (dunkOnIT) [inline app/models/competition.rb:472]: cursory look suggests that this should work, although it's causing a test to fail. I can try debug it in the next day or so

**Review by dunkOnIT — APPROVED** [review]
LGTM

@danieljames-dj would you be able to review/approve as well? I think it's straightforward enough to merge without Gregor's approval, but would appreciate a second set of eyes

**Review by danieljames-dj — APPROVED** [review]
Looks good to me as well :-) 
I'll go ahead and merge this as the change looks pretty safe for me.

**Pull request merged**